### PR TITLE
[MU4] Audio project structure changed

### PR DESCRIPTION
**Pull request by igorkorsukov**
It was: 
* framework/audio/engine 
  *  framework/audio/engine/3rdparty
* framework/audio/midi
  *  framework/audio/midi/3rdparty

Now:
* framework/audio/ 
* framework/midi
* thirdparty/...

This simplifies the structure of the project, makes it more correct, both with regard to audio and to midi.

I like placing `thirdparty` in the module directory, but in this case, it is more difficult to do automation tools, for example, automatically correct and check the code style

### CODE REVIEW

File: main/CMakeLists.txt [hunk 1]
```diff
@@ -87,8 +87,8 @@ if (BUILD_UI_MU4)
         context
         shortcuts
         workspace
-        audio_engine
-        audio_midi
+        mu4_audio
```
Comment (vpereverzev) [inline main/CMakeLists.txt:90]: Is there any naming conflict? Like another audio module or something?
I mean why do we need prefix "mu4"?

Comment (igorkorsukov) [reply main/CMakeLists.txt:90]: Yes, your guess is correct, there is a name conflict.
Therefore, the prefix "mu4" is temporarily added

**Comment by vpereverzev** [comment]
IIRC they have common 3rdparty dependencies, right?
Maybe it's better to keep it like this?

framework/audio:
- engine
- midi
- 3rdparty

**Comment by igorkorsukov** [comment]
> IIRC they have common 3rdparty dependencies, right?
> Maybe it's better to keep it like this?
> 
> framework/audio:
> 
> * engine
> * midi
> * 3rdparty


They have different dependencies
And in this case, it is more difficult to do code verification tools.

File: main/CMakeLists.txt (hunk 1 above)
Comment (psmokotnin) [reply main/CMakeLists.txt:90]: I'd like to suggest to use the prefix "framework" or "fw".

**Pull request merged**